src/libsystemd-network/lldp-neighbor.c

```diff
@@ -793,3 +760,31 @@ int sd_lldp_neighbor_get_timestamp(sd_lldp_neighbor *n, clockid_t clock, uint64_
         *ret = triple_timestamp_by_clock(&n->timestamp, clock);
         return 0;
 }
+
+int lldp_neighbor_build_json(sd_lldp_neighbor *n, JsonVariant **ret) {
+        const char *chassis_id = NULL, *port_id = NULL, *port_description = NULL,
+                *system_name = NULL, *system_description = NULL;
+        uint16_t cc = 0;
+        bool valid_cc;
+
+        assert(n);
+        assert(ret);
+
+        (void) sd_lldp_neighbor_get_chassis_id_as_string(n, &chassis_id);
+        (void) sd_lldp_neighbor_get_port_id_as_string(n, &port_id);
+        (void) sd_lldp_neighbor_get_port_description(n, &port_description);
+        (void) sd_lldp_neighbor_get_system_name(n, &system_name);
+        (void) sd_lldp_neighbor_get_system_description(n, &system_description);
+
+        valid_cc = sd_lldp_neighbor_get_enabled_capabilities(n, &cc);
```

Comment:
Trying to figure out why packets with all the bits set no longer show any enabled capabilities I ended up here. Looks like `valid_cc` is 0 when `n->has_capabilties` is true so 
```c
JSON_BUILD_PAIR_CONDITION(valid_cc, "EnabledCapabilities", JSON_BUILD_UNSIGNED(cc))))
```
doesn't seem to work as expected. The old condition in `networkctl` was
```c
                        if (sd_lldp_neighbor_get_enabled_capabilities(n, &cc) >= 0) {
                                capabilities = lldp_capabilities_to_string(cc);
...
```

Packets were legit:
```sh
        Capabilities: 0xffff
            .... .... .... ...1 = Other: Capable
            .... .... .... ..1. = Repeater: Capable
            .... .... .... .1.. = Bridge: Capable
            .... .... .... 1... = WLAN access point: Capable
            .... .... ...1 .... = Router: Capable
            .... .... ..1. .... = Telephone: Capable
            .... .... .1.. .... = DOCSIS cable device: Capable
            .... .... 1... .... = Station only: Capable
        Enabled Capabilities: 0xffff
            .... .... .... ...1 = Other: Capable
            .... .... .... ..1. = Repeater: Capable
            .... .... .... .1.. = Bridge: Capable
            .... .... .... 1... = WLAN access point: Capable
            .... .... ...1 .... = Router: Capable
            .... .... ..1. .... = Telephone: Capable
            .... .... .1.. .... = DOCSIS cable device: Capable
            .... .... 1... .... = Station only: Capable
```
so they certainly should have capabilities :-)

Reply:
Ouch!